internal/controller/backup_controller.go

```diff
@@ -197,7 +197,7 @@ func (r *BackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		}
 		contextLogger.Debug("Found pod for backup", "pod", pod.Name)
 
-		if !utils.IsPodReady(*pod) {
+		if !utils.IsPodReady(*pod) || !utils.PodHasContainerStatuses(*pod) {
```

Comment:
I think this added condition is not going to be useful.
How can a Pod be ready and not have any container statuses?